editor/src/components/canvas/canvas-strategies/strategies/absolute-move-strategy.spec.browser2.tsx

```diff
@@ -163,6 +184,420 @@ function positioningFromCss(css: CSSStyleDeclaration) {
+      const editor = await renderTestEditorWithCode(
+        makeTestProjectCodeWithSnippet(
+          `
+          <View style={{ ...(props.style || {}) }} data-uid='aaa'>
+            <View
+              style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 50, top: 50, bottom: 250, right: 200 }}
+              data-uid='bbb'
+              data-testid='bbb'
+            />
+          </View>
+        `,
+        ),
+        'await-first-dom-report',
+      )
+      const targetElement = EP.fromString(
+        `${BakedInStoryboardUID}/${TestSceneUID}/${TestAppUID}:aaa/bbb`,
+      )
+
+      await selectComponentsForTest(editor, [targetElement])
+
+      await dragByPixels(editor, windowPoint({ x: 25, y: 10 }), 'bbb', shiftModifier)
+
+      expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+        makeTestProjectCodeWithSnippet(
+          `
+          <View style={{ ...(props.style || {}) }} data-uid='aaa'>
+            <View
+              style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 75, top: 50, bottom: 250, right: 175 }}
+              data-uid='bbb'
+              data-testid='bbb'
+            />
+          </View>
+        `,
+        ),
+      )
+    })
+    it('works with a TL pinned absolute element with child', async () => {
+      const editor = await renderTestEditorWithCode(
+        makeTestProjectCodeWithSnippet(
+          `
+          <View style={{ ...(props.style || {}) }} data-uid='aaa'>
+            <View
+              style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 50, top: 50, width: 250, height: 300 }}
+              data-uid='bbb'
+              data-testid='bbb'
+            >
+              <View
+                style={{ backgroundColor: '#0091FFAB', position: 'absolute', left: 100, top: 100, width: 250, height: 300 }}
+                data-uid='ccc'
+              />
+            </View>
+          </View>
+        `,
+        ),
+        'await-first-dom-report',
+      )
+      const targetElement = EP.fromString(
+        `${BakedInStoryboardUID}/${TestSceneUID}/${TestAppUID}:aaa/bbb`,
+      )
+
+      await selectComponentsForTest(editor, [targetElement])
+
+      await dragByPixels(editor, windowPoint({ x: 15, y: 15 }), 'bbb')
+
+      expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+        makeTestProjectCodeWithSnippet(
+          `
+          <View style={{ ...(props.style || {}) }} data-uid='aaa'>
+            <View
+              style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 65, top: 65, width: 250, height: 300 }}
+              data-uid='bbb'
+              data-testid='bbb'
+            >
+              <View
+                style={{ backgroundColor: '#0091FFAB', position: 'absolute', left: 100, top: 100, width: 250, height: 300 }}
+                data-uid='ccc'
+              />
+            </View>
+          </View>
+`,
+        ),
+      )
+    })
+  })
```

Comment:
I would suggest to refactor these to be table tests, but not necessarily for now :)